### PR TITLE
cray-libsci: initial commit

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: var/spack/repos/builtin/packages/cray-libsci/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    version("16.03.1")
+
+    provides("blas")
+    provides("lapack")
+    provides("scalapack")
+
+    # NOTE: Cray compiler wrappers already include linking for the following
+    @property
+    def blas_libs(self):
+        return LibraryList([self.prefix.lib])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cray-libsci/package.py]: LibraryList takes a list of library files, not directories.

File: var/spack/repos/builtin/packages/cray-libsci/package.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+from llnl.util.filesystem import LibraryList
+from spack import *
+
+
+class CrayLibsci(Package):
+    """The Cray Scientific Libraries package, LibSci, is a collection of
+    numerical routines optimized for best performance on Cray systems."""
+
+    homepage = "http://www.nersc.gov/users/software/programming-libraries/math-libraries/libsci/"
+    url      = "http://www.nersc.gov/users/software/programming-libraries/math-libraries/libsci/"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cray-libsci/package.py]: I would remove this URL if you don't plan on fetching any versions.

Comment (junghans) [reply var/spack/repos/builtin/packages/cray-libsci/package.py]: Having no `url` breaks things.

File: var/spack/repos/builtin/packages/cray-libsci/package.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    version("16.03.1")
+
+    provides("blas")
+    provides("lapack")
+    provides("scalapack")
+
+    # NOTE: Cray compiler wrappers already include linking for the following
+    @property
+    def blas_libs(self):
+        return LibraryList(os.path.join(self.prefix.lib, 'libsci.so'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cray-libsci/package.py]: Is os imported?

Comment (junghans) [reply var/spack/repos/builtin/packages/cray-libsci/package.py]: Good point, added.

File: var/spack/repos/builtin/packages/cray-libsci/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/cray-libsci/package.py]: @tgamblin @tldahlgren I thought we fixed this bug.

**Pull request merged**